packages/foundry/contracts/prediction-market/lib/PredictionMarketLib.sol

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { 
+    PredictionMarket,
+    Side 
+} from '../Types.sol';
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+library PredictionMarketLib { 
+
+    /**
+     * @notice Prediction market has zero token balance for one of both pair tokens
+     */
+    error ZeroTokenBalance();
+
+    /**
+     * @notice Prediction market has insufficent liquidity to execute swap
+     */
+    error CannotSwapInMarket();
```

Comment:
Consider renaming this to `InsufficientLiquidity` or something more descriptive, if that's the only reason it can fail.